### DISCUSSION TITLE
WithLogging() ??

### DISCUSSION THREAD

**Discussion by rmsimpson** [open-telemetry/opentelemetry-dotnet#4653] [Q&A] [answered]
Is there a reason this method is marked internal and not public?

https://github.com/open-telemetry/opentelemetry-dotnet/blob/b5e836152a63aa59c3539171f115f184f7db7482/src/OpenTelemetry.Extensions.Hosting/OpenTelemetryBuilder.cs#L149

I was hoping adding logging with OTLP would be as simple as this code, but even if WithLogging was public, this code still wouldn't work.
None of the logging variations are calling my OtlpExporterOptions configurator, either, so it's reverting to localhost:4317.

Am I totally doing this wrong?

```
services.AddOptions<OtlpExporterOptions>().Configure<IHostServices>((opts, svc) => ConfigureOTLP(opts, svc));

services.AddOpenTelemetry()
	.WithTracing(builder => builder
	.AddOtlpExporter()
	.AddConsoleExporter()
	.ConfigureResource(builder => ConfigureResource(builder))
	)
	.WithMetrics(builder => builder
	.AddOtlpExporter()
	.AddConsoleExporter()
	.ConfigureResource(builder => ConfigureResource(builder))
	)
	.WithLogging(builder => builder
	.AddOtlpExporter()
	.AddConsoleExporter()
	.ConfigureResource(builder = ConfigureResource(builder))
	);


```

**Comment by aygalinc** [comment, marked as answer]
Logging with OTEL is not configured the same way as metrics or trace. 
Probably due to the fact that logging was existing in many Framework before Otel exist. 
So in asp.net they hook the configuration like this : 
    public static IServiceCollection ConfigureOtelLogging(this IServiceCollection services) =>
        
        services.AddLogging(
            it =>
            {
                it.ClearProviders();
                it
                    .AddOpenTelemetry(
                        options =>
                            options.AddConsoleExporter()
                                .IncludeScopes = true);
            });

**Reply by lvde0** [reply]
@aygalinc Seems that `WithLogger` was made public? Can/should we use it now?

**Reply by vishweshbankwar** [reply]
@lvde0 - Yes - WithLogging is part of stable API now. Available starting 1.9.0 version of SDK. https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/RELEASENOTES.md#190
